src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/dataset_retriever.py

```diff
@@ -57,25 +60,56 @@ def from_url(url: str, starting_index: int, length: int) -> List[Dict[str, Any]]
         return formatted_rows
 
     @staticmethod
-    def from_file(file_path: Path) -> List[Dict[str, str]]:
-        with open(file_path, "r") as file:
-            data = [load_json_str(line) for line in file]
+    def from_file(file_path: Path, output_format: OutputFormat) -> List[Dict[str, str]]:
+        contents = DatasetRetriever._load_file_content(file_path)
+
+        dataset = []
+        for content in contents:
+            data = {"text_input": content.get("text_input", "")}
+
+            if output_format == OutputFormat.OPENAI_VISION:
+                img_filename = content.get("image", "")
+                encoded_img = DatasetRetriever._encode_image_to_base64(img_filename)
+                data["image"] = encoded_img
+
+            dataset.append(data)
+        return dataset
 
-            for item in data:
-                if not isinstance(item, dict):
+    @staticmethod
+    def _load_file_content(file_path: Path) -> List[Dict[str, str]]:
+        contents = []
+        with open(file_path, "r") as file:
+            for line in file:
+                content = utils.load_json_str(line)
+                if not isinstance(content, dict):
                     raise GenAIPerfException(
                         "File content is not in the expected format."
                     )
-                if "text_input" not in item:
-                    raise GenAIPerfException(
-                        f"Missing 'text_input' field in file item: {item}"
-                    )
-                if len(item) != 1:
+                if "text_input" not in content:
                     raise GenAIPerfException(
-                        f"Field other than 'text_input' field found in file item: {item}"
+                        f"Missing 'text_input' field in file content: {content}"
                     )
+                contents.append(content)
 
-            return [{"text_input": item["text_input"]} for item in data]
+        return contents
+
+    @staticmethod
+    def _encode_image_to_base64(filename: str) -> str:
```

Comment:
Undecided on whether we want this here or in a utils class.

Reply:
Can we rename this, since it's getting image data rather than encoding it (that's done with the utils function)?

Reply:
sure, renamed it to `_read_image_content`